### DISCUSSION TITLE
Who generates my stress_xy, ...xz, ...?

### DISCUSSION THREAD

**Discussion by jmeier** [idaholab/moose#27819] [Q&A Modules: Solid mechanics] [answered]
Dear Community,

In my Moose input file I use the "Physics/SolidMechanics/QuasiStatic" block. The model itself works fine. 

I only request stress_xx, _yy, and _zz to be present as AuxVariable but Moose is generating all the others too. Moose writes:
```
Auxiliary System:
  Num DOFs:                270028
  Num Local DOFs:          17325
  Variables:               { "stress_xx" "stress_yy" "stress_zz" "stress_xy" "stress_xz" ... "stress_yz" 
                             "stress_zx" "stress_zy" "p" "q" } 
  Finite Element Types:    "MONOMIAL" 
  Approximation Orders:    "CONSTANT" 
```

I'd like to restrict Moose to have only `Variables: { "stress_xx" "stress_yy" "stress_zz" "p" "q" } `
My questions are: Why Moose does this and how can I restrict the AuxVariables generated?

My "Physics/SolidMechanics/QuasiStatic" block looks like (this is my only block of this type, no includes in my file...):
```
[Physics]
  [SolidMechanics]
    [QuasiStatic]
      [.all]
        strain = SMALL
        incremental = true
        add_variables = true
        eigenstrain_names = ini_stress
        generate_output = 'stress_xx stress_yy stress_zz'
      []
    []
  []
[]
```
There are some additional AuxVariables and AuxKernels:
```
[AuxVariables]
  [p]
    order = CONSTANT
    family = MONOMIAL
  []
  [q]
    order = CONSTANT
    family = MONOMIAL
  []
[]

[AuxKernels]
  [p]
    type = RankTwoScalarAux
    rank_two_tensor = stress
    variable = p
    scalar_type = hydrostatic
  []
  [q]
    type = RankTwoScalarAux
    rank_two_tensor = stress
    variable = q
    scalar_type = vonmisesStress
  []
[]
```

Output-block looks like:
```
[Outputs]
  exodus = true
[]
```

**Comment by GiudGiud** [comment, marked as answer]
Hello

This could be modified in the QuasiStaticSolidMechanics. We simply havent worked on limiting the output. 
Feel free to make a pull request if you really need this

Guillaume

**Reply by jmeier** [reply]
Dear Guillaume, 
Thank you for the clarification! I checked the code of [QuasiStaticSolidMechanicsPhysics](https://github.com/idaholab/moose/blob/0445d490c50428b8ce3cbf4f27e1b017456306ff/modules/solid_mechanics/src/physics/QuasiStaticSolidMechanicsPhysics.C), but realized that the restriction of the output must be done by someone more experienced.
Jörg
